### PR TITLE
Upgrade/ksl550 upgrade wagtail

### PR DESCRIPTION
**Pull request by davidu1975**
Upgrade great-cms to Wagtail 4.2.2 to fix vulnerabilities in 2.15.6
 
### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/TICKET_ID_HERE
- [x] Jira ticket has the correct status.
- [x] A clear/description pull request messaged added.

### Housekeeping

- [x] Upgraded any vulnerable dependencies.
- [x] Python requirements have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

**Review by mforner13 — COMMENTED** [review]
Looks great! Just a couple of v minor comments.
It would also be good to clean up commits before merging.

File: tests/unit/core/test_wagtail_hooks.py [hunk 1]
```diff
@@ -429,14 +438,19 @@ def test__update_data_for_appropriate_version(domestic_homepage, rf, force_updat
...
 
     wagtail_hooks._update_data_for_appropriate_version(
         page=detail_page, force_page_update=force_update, data_to_update={'title': 'Dummy Title'}
     )
 
     revision.refresh_from_db()
-    assert json.loads(revision.content_json)['title'] == 'Dummy Title'
+
+    if isinstance(revision.content, str):
+        jason_content = json.loads(revision.content)
```
[the remark this answers is not in the document]
Comment (davidu1975) [reply tests/unit/core/test_wagtail_hooks.py:450]: changed

File: core/wagtail_hooks.py [hunk 2]
```diff
@@ -79,9 +79,13 @@ def _update_data_for_appropriate_version(page: Page, force_page_update: bool, da
     (eg, because the Live page has just been created so has no
     unpublished changes, but we still want to update it with data_to_update)
     """
-
     latest_revision = page.get_latest_revision()
-    latest_revision_json = json.loads(latest_revision.content_json)
+
+    if isinstance(latest_revision.content, str):
+        jason_object = json.loads(latest_revision.content)
```
[the remark this answers is not in the document]
Comment (davidu1975) [reply core/wagtail_hooks.py:85]: changed

[the remark this answers is not in the document]
Comment (davidu1975) [reply core/wagtail_hooks.py:85]: changed

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -43,13 +43,13 @@ repos:
...
             pass_filenames: true
             require_serial: true
-          - id: pii_secret_file_content
-            files: ''
-            language: python
-            args: [--exclude=pii-secret-exclude.txt]
-            pass_filenames: true
-            require_serial: true
-            exclude: core/static/fonts/
+          # - id: pii_secret_file_content
```
[the remark this answers is not in the document]
Comment (davidu1975) [reply .pre-commit-config.yaml]: uncommented out

**Review by mforner13 — APPROVED** [review]
Nice one 👍

**Pull request merged**